autogpts/autogpt/autogpt/agents/base.py

```diff
@@ -145,47 +136,102 @@ class BaseAgentSettings(SystemSettings):
     """(STATE) The action history of the agent."""
 
 
-class BaseAgent(Configurable[BaseAgentSettings], ABC):
-    """Base class for all AutoGPT agent classes."""
+class AgentMeta(type):
+    def __call__(cls, *args, **kwargs):
+        # Create instance of the class (Agent or BaseAgent)
+        instance = super().__call__(*args, **kwargs)
+        # Automatically collect modules after the instance is created
+        instance._collect_components()
+        return instance
+
+
+class CombinedMeta(ABCMeta, AgentMeta):
+    def __new__(cls, name, bases, namespace, **kwargs):
+        return super().__new__(cls, name, bases, namespace, **kwargs)
```

Comment:
This is necessary so the Agent can inherit from two metaclasses

Reply:
Looking at this again, I wonder if `BaseAgent(ABC, metaclass=AgentMeta)` could also work. That would negate the need for the `AgentABCMeta` class.

Reply:
Done! I also had to change `class AgentMeta(type)` -> `class AgentMeta(ABCMeta)`